### DISCUSSION TITLE
Interface disappearing, app crashing?

### DISCUSSION THREAD

**Discussion by neotmhy** [ruittenb/Spaceman#1] [General]
Hi ruitenb, I notice you have the most active fork of the Spaceman app. The original version wasn't working, so I thought I'd try yours. I've been trying out (Spaceman-1.6.0-R-alpha), and it looks promising. For some reason, the app has stopped working for me though. At first I was able to add names to my spaces, but before I could try using them, the app appeared to silently crash. It no longer appeared in the Activity Monitor. The only sign of it was a reference to it visible from `ps -ef`. I killed that and tried restarting it, but got the same behaviour. I then tried installing the latest release version (Spaceman-1.5.9-R), but its interface won't load up either. Any suggested workarounds? I'm running macOS Sonoma 14.5 in an Intel MacBook Pro 16-Inch, 2019.

**Comment by ruittenb** [comment]
Hello neotmhy,

For some of the app defaults the data format has changed. This causes the app to crash. It is a known problem. I hope that 1.6.1-R-alpha solves that problem. Could you please try that?

If that does not work, you could try the following command:

`defaults delete dev.jaysce.Spaceman`

Please let me know if that works for you.

**Comment by ruittenb** [comment]
@neotmhy : Version 1.6.6 is the latest stable version, and it should fix your problem.
Please try https://github.com/ruittenb/Spaceman/releases/tag/v1.6.6

**Reply by neotmhy** [reply]
Hi ruittenb,

Thank you for your reply. I just installed v1.6.6 version of Spaceman, which triggered an update to v1.6.17. The interface is still not showing up, other than when it triggered the update. In order to see that I ran it rom a Terminal prompt with '/Applications/Spaceman.app/Contents/MacOS/Spaceman'. Even doing it that way still doesn't show the interface following the update. I tried your suggested defaults deletion, but that didn't seem to help. One improvement is that now when I launch it, it does show up in Activity Monitor, and can a least be killed from there.

**Reply by ruittenb** [reply]
Hello neotmhy,
Could you please try the following steps:
- kill Spaceman
- (in terminal) defaults delete dev.jaysce.Spaceman
- (in terminal) defaults delete dev.ruittenb.Spaceman
- launch Spaceman
Also, please make sure that your menu bar at the top of the screen has enough room for all the space icons.

**Comment by neotmhy** [comment]
Hi ruittenb,

Your comment about having enough space for the icons was helpful. I'm using Bartender to manage this and found the Spaceman icons were hidden. Once I placed them in the Shown area, I was then able to get to the Spaceman interface and was able to rename the spaces. Things work as expected when choosing to display just the space names, but when I try to display both names and numbers, the icons disappear, even if tapping in an attempt to see those that might not fit in the available icon display area. In any case, it is now working and I thank you!

**Reply by ruittenb** [reply]
Yes, Spaceman makes one composite icon to display in the menu bar. As a consequence, if the icon doesn't fit, MacOS decides to hide it completely. This functionality/quirk was also present in the original Spaceman.

Great to hear that this has been resolved!

**Reply by ruittenb** [reply]
@neotmhy Could you mark this issue as "answered"?
